### ISSUE TITLE
register_shutdown_function() causes major memory leak during testing

### ISSUE THREAD

**Issue by ryancastle** [bugsnag/bugsnag-symfony#77]
### Expected behavior
Symfony tests should run under phpunit without major memory leaks.

Alternatively, provide a bootstrapping process which can be modified for testing. Currently, the hard-coding of the `Bugsnag\Client` class makes that very difficult.

### Observed behavior
Each request/sub-request during testing adds 2MB-4MB of permanent memory consumption.

### Steps to reproduce
```
class TestCase extends \Symfony\Bundle\FrameworkBundle\Test\WebTestCase 
{
    public function testMemory()
    {
        echo memory_get_usage() . PHP_EOL;
        $client->request('GET', '/');
        echo memory_get_usage() . PHP_EOL;
        $client->request('GET', '/');
        $client->request('GET', '/');
        $client->request('GET', '/');
        $client->request('GET', '/');
        $client->request('GET', '/');
        echo memory_get_usage() . PHP_EOL;
    }
}

```
### Version
1.4.0/1.5.0

### Additional information
I have confirmed that removing the `register_shutdown_function([$this, 'flush']);` from the `Client` constructor resolves the issue. Unfortunately, that encloses the state of `$this` until the process shuts down, which on large test suites is usually when it crashes from memory problems.

**Comment by ryancastle** [comment]
This is a functional work-around stub for testing, but very sad.

**config_test.yml** or **test/bugsnag.yaml**
```
bugsnag:
    client: App\Bugsnag\TestClient
    factory: App\Bugsnag\TestFactory
```

**TestClient**
```
use Bugsnag\Breadcrumbs\Recorder;
use Bugsnag\Client;
use Bugsnag\Configuration;
use Bugsnag\HttpClient;
use Bugsnag\Pipeline;
use Bugsnag\Request\BasicResolver;
use Bugsnag\Request\ResolverInterface;
use Bugsnag\SessionTracker;
use GuzzleHttp\ClientInterface;

class TestClient extends Client
{
    public function __construct(
        Configuration $config,
        ResolverInterface $resolver = null,
        ClientInterface $guzzle = null
    ) {
        $this->config = $config;
        $this->resolver = $resolver ?: new BasicResolver();
        $this->recorder = new Recorder();
        $this->pipeline = new Pipeline();
        $this->http = new HttpClient($config, $guzzle ?: static::makeGuzzle());
        $this->sessionTracker = new SessionTracker($config);
    }
}
```

**TestFactory**
```
use Bugsnag\BugsnagBundle\DependencyInjection\ClientFactory;
use Bugsnag\BugsnagBundle\Request\SymfonyResolver;
use Bugsnag\Client;
use Bugsnag\Configuration;

class TestFactory extends ClientFactory
{
    /**
     * @return Client
     */
    public function make()
    {
        return new TestClient(new Configuration(''), new SymfonyResolver(), new \GuzzleHttp\Client());
    }
}
```

**Comment by mattdyoung** [comment]
Fixed in [v1.6.0](https://github.com/bugsnag/bugsnag-symfony/releases/tag/v1.6.0)
